pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -482,6 +486,15 @@ public void updateAll() {
     // there by each broker via updateLocalBrokerData.
     private void updateAllBrokerData() {
         final Set<String> activeBrokers = getAvailableBrokers();
+        Collection<String> newBookies = CollectionUtils.subtract(activeBrokers, knownBrokers);
+        knownBrokers.addAll(newBookies);
+        Collection<String> deadBrokers = CollectionUtils.subtract(knownBrokers, activeBrokers);
+        knownBrokers.removeAll(deadBrokers);
+        for (String broker: deadBrokers) {
+            final String timeAverageZPath = TIME_AVERAGE_BROKER_ZPATH + "/" + broker;
+            timeAverageBrokerDataCache.delete(timeAverageZPath).join();
```

Comment:
Does this throw exception if path does not exists?

Reply:
done @Jason918